fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/WitnessMessageHandler.java

```diff
@@ -0,0 +1,116 @@
+package com.github.dedis.popstellar.utility.handler;
+
+import android.util.Log;
+import com.github.dedis.popstellar.model.Lao;
+import com.github.dedis.popstellar.model.PendingUpdate;
+import com.github.dedis.popstellar.model.WitnessMessage;
+import com.github.dedis.popstellar.model.data.LAORepository;
+import com.github.dedis.popstellar.model.network.method.message.MessageGeneral;
+import com.github.dedis.popstellar.model.network.method.message.PublicKeySignaturePair;
+import com.github.dedis.popstellar.model.network.method.message.data.message.WitnessMessageSignature;
+import com.github.dedis.popstellar.utility.security.Signature;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Witness messages handler class
+ */
+public class WitnessMessageHandler {
+
+  private static final String TAG = WitnessMessage.class.getSimpleName();
+
+  private WitnessMessageHandler() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Process a WitnessMessageSignature message
+   *
+   * @param laoRepository the repository to access the LAO of the channel
+   * @param channel       the channel on which the message was received
+   * @param senderPk      the public key of the sender
+   * @param message       the message that was received
+   * @return true if the message cannot be processed and false otherwise
+   */
+  public static boolean handleWitnessMessage(LAORepository laoRepository,
+      String channel,
+      String senderPk,
+      WitnessMessageSignature message) {
+    Log.d(TAG, "Received Witness Message Signature Broadcast");
+    String messageId = message.getMessageId();
+    String signature = message.getSignature();
+
+    byte[] senderPkBuf = Base64.getUrlDecoder().decode(senderPk);
+    byte[] signatureBuf = Base64.getUrlDecoder().decode(signature);
+
+    // Verify signature
+    if (!Signature.verifySignature(messageId, senderPkBuf, signatureBuf)) {
+      return false;
+    }
+
+    if (laoRepository.getMessageById().containsKey(messageId)) {
+      // Update the message
+      MessageGeneral msg = laoRepository.getMessageById().get(messageId);
+      msg.getWitnessSignatures().add(new PublicKeySignaturePair(senderPkBuf, signatureBuf));
```

Comment:
Nothing for you to do in this PR, but this bunch of code really makes me think the Java front-end could stand to benefit from some of the things we introduced in fe1-web, such as having `PublicKey`, `Hash` and `Signature` classes, abstracting away the representation/encoding (bytes vs string).

The code would then look like this, don't you think it reads better?
```suggestion
      String channel,
      PublicKey senderPk,
      WitnessMessageSignature message) {
    Log.d(TAG, "Received Witness Message Signature Broadcast");
    Hash messageId = message.getMessageId();
    Signature signature = message.getSignature();

    // Verify signature
    if (!signature.verify(messageId, senderPk)) {
      return false;
    }

    if (laoRepository.getMessageById().containsKey(messageId)) {
      // Update the message
      MessageGeneral msg = laoRepository.getMessageById().get(messageId);
      msg.getWitnessSignatures().add(new PublicKeySignaturePair(senderPk, signature));
```